src/libraries/Native/Unix/System.Native/pal_signal.c

```diff
@@ -276,6 +494,27 @@ static bool CreateSignalHandlerThread(int* readFdPtr)
 
 int32_t InitializeSignalHandlingCore()
 {
+    size_t signalMax = (size_t)GetSignalMax();
+    g_origSigHandler = (struct sigaction*)calloc(sizeof(struct sigaction), signalMax);
+    g_handlerIsInstalled = (bool*)calloc(sizeof(bool), signalMax);
+    g_hasPosixSignalRegistrations = (bool*)calloc(sizeof(bool), signalMax);
+    if (g_origSigHandler == NULL ||
+        g_handlerIsInstalled == NULL ||
+        g_hasPosixSignalRegistrations == NULL)
+    {
+        free(g_origSigHandler);
+        free(g_handlerIsInstalled);
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wcast-qual"
```

Comment:
We have tried to keep the runtime compilable with GCC. Does GCC compile the code fine? 
Actually, I've tried to compile this construct using the compiler explorer and I didn't get any warnings with the compiler options we use. The (void*) cast should be sufficient. Isn't this a remainder of some previous form of the code?

Reply:
When I use: `free((void*)g_hasPosixSignalRegistrations);` I get:
```
 /home/tmds/repos/runtime/src/libraries/Native/Unix/System.Native/pal_signal.c:531:21: error: cast from 'volatile bool *' to 'void *' drops volatile qualifier [-Werror,-Wcast-qual]
          free((void*)g_hasPosixSignalRegistrations);
```
The error doesn't even make sense to me because `g_hasPosixSignalRegistrations` is not a `volatile` pointer, but the array elements are.
I'm using clang `11.0.0-3` on Fedora 33.

Reply:
One way to fix it without the pragmas is this:
```c
free((void*)(size_t)g_hasPosixSignalRegistrations);
```

Reply:
That looks better.